packages/mds-db/policies.ts

```diff
@@ -106,9 +133,24 @@ export async function readPolicy(policy_id: UUID): Promise<Policy> {
   throw new NotFoundError(`policy_id ${policy_id} not found`)
 }
 
+async function throwIfRulesAlreadyExist(policy: Policy) {
+  await Promise.all(
+    policy.rules.map(async rule => {
```

Comment:
I don't think this method is working as intended -- because you declare the method you're mapping across all of the entries to be `async` and you're `await`ing inside it, the method is behaving synchronously, and the wrapping `Promise.all` is not actually doing anything. This should be refactored to prepare an array of promises with the `readPolicies` query in it, await a `Promise.all` to fetch them, then iterate through and check each of the rules in your policy against any that also have the corresponding rule.